### DISCUSSION TITLE
OpenGlass Legacy v1.0 beta

### DISCUSSION THREAD

**Discussion by ALTaleX531** [ALTaleX531/OpenGlass#9] [Announcements]
### Introduction
OpenGlass Legacy is a dcomp-independent implementation which is designed to provide more efficient performance.
> [!NOTE]
> This is a beta version and when you use it you need to know the following things.
> 1. This branch is not yet complete, so some features may not be available.
> 2. You may encounter a variety of bugs.
> 3. You can remain both this implementation and the default implementation on your machine, but remember not to run them at the same time. 
### Known issue
When AccentState is 3, accent causes artifacts when the blurred area is updated.
> [!TIP]
> If you want to give feedback, please do it in Discussion.

**Full Changelog**: https://github.com/ALTaleX531/OpenGlass/compare/v1.2...v1.0-legacy-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/ALTaleX531/OpenGlass/releases/tag/v1.0-legacy-beta'>OpenGlass Legacy v1.0 beta</a>.</em>

**Comment by wiktorwiktor12** [comment]
This looks great, and it seems promising for performance!
I'm curious, are there any plans to use a hlsl shader instead of effects? Could be very cool for customization (especially because of that one longhorn shader being hlsl) and perhaps performance (could be wrong, untested im assuming).

**Reply by ALTaleX531** [reply]
Theoretically this is possible since I'm using native Direct2D for rendering, but using the hlsl shader may be worse in performance since I'm using the highly optimized Direct2D internal blur. For now, let's get all the bugs fixed before thinking about adding new features.

**Reply by wiktorwiktor12** [reply]
Yes I agree, that sounds like a good idea

**Comment by aubymori** [comment]
@ALTaleX531 Here's the issue I mentioned in #5. Sorry for recording with my phone, OBS seemed to mess with the shader (which I'll post too).
HL2 with `cl_showpos 1` in the background so you can easily see when it's blurred and unblurred.

https://github.com/ALTaleX531/OpenGlass/assets/44238627/1f20deda-4641-4383-a734-932ca7053656

Here's what happens with OBS:

https://github.com/ALTaleX531/OpenGlass/assets/44238627/cd12ba4c-14cd-4d9a-82e2-f00b9b8f8366

**Reply by aubymori** [reply]
Turns out what OBS did wasn't actually recorded... here it is:

https://github.com/ALTaleX531/OpenGlass/assets/44238627/a0d98d5c-2f27-4251-a4d8-f0004b52a5bc

**Reply by ALTaleX531** [reply]
[OpenGlass.zip](https://github.com/user-attachments/files/16039146/OpenGlass.zip)
Then try this version, I've refactored some of the code and this should fix the issues.

**Reply by aubymori** [reply]
> [OpenGlass.zip](https://github.com/user-attachments/files/16039146/OpenGlass.zip) Then try this version, I've refactored some of the code and this should fix the issues.

Works great! The issue with blur disappearing is still here (mostly when switching to a non-maximized window) but is less common. The other issue is gone completely.

**Reply by ALTaleX531** [reply]
This is a really weird issue, and currently I'm still not sure what causes it and how to reproduce it. It may not be solved at the moment until new clues appear.

**Comment by akaydev-coder** [comment]
Sorry for Offtopic: are you active on ko-fi.com? Send a message and donation ( I wish I could send more )

**Reply by ALTaleX531** [reply]
Oh, you sent me a message and a donation? Thank you very much! I blocked all emails from ko-fi about a couple months ago because it kept sending ads, but I didn't expect anyone to be able to find my account and I don't plan on using it.

Regarding the message you sent, I only used IDA to reverse engineer dwmcore.dll and udwm.dll and simply scanned the hooks that glass8 made to dwm, and I've never established contact with BigMuscle, so I'm not really sure what's going on with him right now.

**Reply by akaydev-coder** [reply]
Thanks for reply and explanation.
Re KoFi I was happy to find a platform where I can send a donation. Usually PayPal ( don't like it ) but I did not find this.
